### PR TITLE
Selection Range

### PR DESCRIPTION
**Pull request by HannaParsa**

### CODE REVIEW

**Comment by HannaParsa** [comment]
@mahdiHash 
This is the new branch that I will work on.

**Comment by mahdiHash** [comment]
> @mahdiHash This is the new branch that I will work on.

Hi, @HannaParsa.
I'm glad to see you still helping with the translation process.
I'm no longer a maintainer of this repo, so if you need to contact one, that's not me!
Have fun translating :)

P.S. I guess it's OK to translate on any branch you want.

**Comment by HannaParsa** [comment]
@mahdyar 
This is the branch that I will be working on

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Review by mahdyar — CHANGES_REQUESTED** [review]
You still have different line numbers with the English version.

image

They're both 718 lines, but each line should also match the English version. If you're using any formatter, please disable it.

For example you've deleted line number 4 and "Selection and Range" header is now in line number 6 instead of 7:
image

File: 2-ui/99-ui-misc/02-selection-range/article.md [hunk 1]
```diff
@@ -49,10 +48,10 @@ let range = new Range();
 </script>
 ```
 
-در اینجا اولین فرزند `<p>` را می گیریم (این text node است) و موقعیت های متن را در داخل آن مشخص می کنیم:
-
 
+در اینجا اولین فرزند `<p>` را می گیریم (این text node است) و موقعیت های متن را در داخل آن مشخص می کنیم:
 ![](range-hello-1.svg)
+
 ### Selecting element nodes
```
Comment (mahdyar) [inline 2-ui/99-ui-misc/02-selection-range/article.md:55]: Move the new line to after line 52.

File: 2-ui/99-ui-misc/02-selection-range/article.md [hunk 2]
```diff
@@ -32,7 +32,6 @@ let range = new Range();
 
 نکته جالب این است که آرگومان اول `node` در هر دو روش می تواند یک text node یا یelement node عنصر باشد و معنای آرگومان دوم به آن بستگی دارد.
 
-
```
Comment (mahdyar) [inline 2-ui/99-ui-misc/02-selection-range/article.md:35]: Please add this line back as well.

File: 2-ui/99-ui-misc/02-selection-range/article.md [hunk 3]
```diff
@@ -61,7 +60,6 @@ let range = new Range();
 این برای ایجاد محدوده هایی که شامل گره ها به عنوان یک کل هستند، مفید است، نه اینکه در جایی در متن خود متوقف شوند.
 
 به عنوان مثال، ما یک قطعه سند پیچیده تر داریم:
-
```
Comment (mahdyar) [inline 2-ui/99-ui-misc/02-selection-range/article.md:64]: Please add this line back as well.

**Review by mahdyar — APPROVED** [review]

**Review by mahdyar — CHANGES_REQUESTED** [review]
Please review requested changes.

File: 2-ui/99-ui-misc/02-selection-range/article.md [hunk 4]
```diff
@@ -105,17 +104,17 @@ drawHtmlTree(selectPDomtree, 'div.select-p-domtree', 690, 320);
 </script>
 
 
-همانطور که می بینیم، این عبارت دقیقاً از دو فرزند `<p>` با نمایه های  `0`  و `1` تشکیل شده است:
+بیایید یک محدوده برای `"Example: <i>italic</i>"` بسازیم.
 
+همانطور که می بینیم، این عبارت دقیقاً از دو فرزند `<p>` با نمایه های  `0`  و `1` تشکیل شده است:
 ![](range-example-p-0-1.svg)
 
-
```
Comment (mahdyar) [inline 2-ui/99-ui-misc/02-selection-range/article.md:112]: Add this line back.

**Review by mahdyar — CHANGES_REQUESTED** [review]

File: 2-ui/99-ui-misc/02-selection-range/article.md [hunk 5]
```diff
@@ -495,36 +495,36 @@ The exception is some selection methods, that replace the existing selection, su
...
-Properties:
-- `input.selectionStart` -- position of selection start (writeable),
-- `input.selectionEnd` -- position of selection end (writeable),
-- `input.selectionDirection` -- selection direction, one of: "forward", "backward" or "none" (if e.g. selected with a double mouse click),
+عناصر فرم، مانند `input` و textarea [special API for selection](https://html.spec.whatwg.org/#textFieldSelection) را بدون اشیاء `Selection` یا `Range` ارائه می‌کنند. از آنجایی که مقدار ورودی یک متن خالص است، نه HTML، نیازی به چنین اشیایی نیست، همه چیز بسیار ساده تر است.
 
-Events:
-- `input.onselect` -- triggers when something is selected.
-
```
Comment (mahdyar) [inline 2-ui/99-ui-misc/02-selection-range/article.md:507]: Please check these lines as well. and all the lines. the line numbers on both sides for every single line should be the same like this:

image

**Review by mahdyar — CHANGES_REQUESTED** [review]

File: 2-ui/99-ui-misc/02-selection-range/article.md (hunk 5 above)
Comment (HannaParsa) [reply 2-ui/99-ui-misc/02-selection-range/article.md:507]: I think it is correct, for example line 498 and 499 are both the same and also other lines.
@mahdyar

File: 2-ui/99-ui-misc/02-selection-range/article.md [hunk 6]
```diff
@@ -1,23 +1,23 @@
...
-درک اشیاء زیربنایی  `Range` و`Selection` آسان است، و پس از آن برای اینکه آنها را به آنچه می‌خواهید انجام دهند، به هیچ دستور العملی نیاز نخواهید داشت.
+می‌توانید دستور العمل‌هایی برای کارهای رایج در انتهای فصل، در بخش "Summary" بیابید. شاید این نیازهای فعلی شما را پوشش دهد، اما اگر متن کامل را بخوانید، خیلی بیشتر به دست خواهید آورد.
 
+درک اشیاء زیربنایی `Range` و `Selection` آسان است، و پس از آن برای اینکه آنها را به آنچه می‌خواهید انجام دهند، به هیچ دستور العملی نیاز نخواهید داشت.
 ## Range
 
 فهوم اصلی انتخاب [Range](https://dom.spec.whatwg.org/#ranges) است، که اساساً یک جفت "boundary points"است: شروع محدوده و پایان محدوده.
 
-
 یک `Range` object بدون پارامتر ساخته می شود:
```
Comment (mahdyar) [inline 2-ui/99-ui-misc/02-selection-range/article.md:22]: لطفا همه خطوط رو خودتون بررسی کنید حتما دو طرف یکی باشه. اینجا مثلا نیستن.

**Review by mahdyar — CHANGES_REQUESTED** [review]

**Review by mahdyar — APPROVED** [review]
Thank you, @HannaParsa!

**Pull request merged**